### PR TITLE
feat: support github enterprise api

### PR DESCRIPTION
**Pull request by ricardojdsilva87**
# Pull Request

## Proposed Changes

This Pull request is based on the already merged https://github.com/github/evergreen/pull/256 to give support to GitHub Enterprise login using a GitHub App also created on GitHub Enterprise

**Code coverage**

```
---------- coverage: platform darwin, python 3.13.0-final-0 ----------
Name                          Stmts   Miss  Cover   Missing
-----------------------------------------------------------
auth.py                          28      4    86%   50, 80-82
classes.py                       11      0   100%
config.py                        86      0   100%
discussions.py                   15      0   100%
issue_metrics.py                 65     10    85%   92, 110, 124-127, 139, 149, 154, 366
json_writer.py                   64      3    95%   95, 198-199
labels.py                        71      5    93%   55, 78, 94, 114, 118
markdown_helpers.py              11      0   100%
markdown_writer.py              105      5    95%   149, 263, 273, 283, 293
most_active_mentors.py           46     13    72%   97-113, 116-130
search.py                        66     23    65%   46-56, 81-105, 134
time_in_draft.py                 28      0   100%
time_to_answer.py                28      0   100%
time_to_close.py                 38      2    95%   49, 55
time_to_first_response.py        71      3    96%   88-89, 121
time_to_merge.py                 12      0   100%
time_to_ready_for_review.py      15      3    80%   45-49
-----------------------------------------------------------
TOTAL                           760     71    91%

Required test coverage of 80% reached. Total coverage: 90.66%
```

### Author/Contributor

- [x] If documentation is needed for this change, has that been included in this pull request
- [x] run `make lint` and fix any issues that you have introduced
- [x] run `make test` and ensure you have test coverage for the lines you are introducing
- [ ] If publishing new data to the public (scorecards, security scan results, code quality results, live dashboards, etc.), please request review from `@jeffrey-luszcz`

### Reviewer

- [ ] Label as either `fix`, `documentation`, `enhancement`, `infrastructure`, `maintenance`, or `breaking`

### CODE REVIEW

File: markdown_writer.py [hunk 1]
```diff
@@ -185,15 +187,19 @@ def write_to_markdown(
             # Replace any whitespace
             issue.title = issue.title.strip()
 
+            endpoint = ghe.removeprefix("https://") if ghe else "github.com"
```
Comment (jmeridth) [inline markdown_writer.py:190]: Same suggestion from [here](https://github.com/github/contributors/pull/197#discussion_r1819987751).  It might make the changes below more complex.  wdyt?

File: markdown_writer.py [hunk 2]
```diff
@@ -185,15 +187,19 @@ def write_to_markdown(
...
                     f"| {issue.title} | "
-                    f"{issue.html_url.replace('https://github.com', 'https://www.github.com')} |"
+                    f"{issue.html_url}".replace(
+                        f"https://{endpoint}", f"https://www.{endpoint}"
+                    )
+                    + " |"
                 )
             else:
-                file.write(f"| {issue.title} | " f"{issue.html_url} |")
+                file.write(f"| {issue.title} | {issue.html_url} |")
```
Comment (jmeridth) [inline markdown_writer.py:200]: nice catch :tada:

**Review by jmeridth — COMMENTED** [review]
One suggestion up for discussion.  Great catch on the misspellings.  Thank you for adding this feature.

File: markdown_writer.py (hunk 1 above)
Comment (ricardojdsilva87) [reply markdown_writer.py:190]: Hello @jmeridth,
I might be confusing something in this case :)
My idea was that the replace done next and also in some other parts of the code github.com is hardcoded, but we might need to consider that the `ghe` variable is something like `https://yourgithubenterprise.com`.
In this case should't all the references to `github.com` be switched to the `yourgithubenterprise.com` domain?
Again I might be missing something in this case since I couldn't fully test the action against our GHE instance
Basically the main idea here is to use the ghe domain if the user has set it or github.com by default.
Thanks

Comment (jmeridth) [reply markdown_writer.py:190]: This ask isn't enough to block. I understand your code change. I'll do a follow-up PR to express my change.

**Review by jmeridth — APPROVED** [review]

File: markdown_writer.py (hunk 1 above)
Comment (ricardojdsilva87) [reply markdown_writer.py:190]: @jmeridth did some changes also [here](https://github.com/github/contributors/pull/197#discussion_r1820426913).
I will take a look also in this PR
